metric_test.go

```diff
@@ -20,3 +20,30 @@ func BenchmarkSetId(b *testing.B) {
 		metric.SetId()
 	}
 }
+
+func TestTagValidation(t *testing.T) {
+	type testCase struct {
+		tag       []string
+		expecting bool
+	}
+
+	testCases := []testCase{
+		{[]string{"abc=cba"}, true},
+		{[]string{"a="}, false},
+		{[]string{"a!="}, false},
+		{[]string{"=abc"}, false},
+		{[]string{"@#$%!=(*&"}, false},
+		{[]string{"!@#$%=(*&"}, false},
+		{[]string{"@#;$%=(*&"}, false},
+		{[]string{"@#$%=(;*&"}, false},
+		{[]string{"@#$%=(*&"}, true},
+		{[]string{"@#$%=(*&", "abc=!fd", "a===="}, true},
+		{[]string{"@#$%=(*&", "abc=!fd", "a===;="}, false},
+	}
+
+	for _, tc := range testCases {
+		if validateTags(tc.tag) != tc.expecting {
+			t.Fatalf("Testcase %s returned %t, but expected %t", tc.tag, !tc.expecting, tc.expecting)
```

Comment:
it's typical to do "expected" first, "got" afterwards

Reply:
i meant in the Fatalf message

Reply:
ah, ok